packages/mdl-ripple/README.md

```diff
@@ -258,6 +281,60 @@ the same effect.
 </div>
 ```
 
+### Keyboard interaction for custom UI components
+
+Different keyboard events activate different elements. For example, the space key activate buttons, while the enter key activates links. Handling this by sniffing the key/keyCode of an event is brittle and error-prone, so instead we take the approach of using `adapter.isSurfaceActive()`. The
+way in which our default vanilla DOM adapter determines this is by using
+`element.matches(':active')`. However, this approach will _not_ work for custom components that
+the browser does not apply this pseudo-class to.
+
+If you want your component to work properly with keyboard events, you'll have to listen for both `keydown` and `keyup` and set some sort of state that the adapter can use to determine whether or
+not the surface is "active", e.g.
+
+```js
+class MyComponent {
+  constructor(el) {
+    this.el = el;
+    this.active = false;
+    this.ripple_ = new MDLRippleFoundation({
+      // ...
+      isSurfaceActive: () => this.active
+    });
+    this.el.addEventListener('keydown', evt => {
+      if (isSpace(evt)) {
+        this.active = true;
+      }
+    });
+    this.el.addEventListener('keyup', evt => {
+      if (isSpace(evt)) {
+        this.active = false;
+      }
+    });
+  }
+}
+```
+
+### Specifying known element dimensions
+
+If you asynchronously load style resources, such as loading stylesheets dynamically via scripts
+or loading fonts, then `adapter.getClientRect()` may by default return _incorrect_ dimensions when
+the ripple foundation is initialized. For example, if you put a ripple on an element that uses an
+icon font, and the size of the icon font isn't specified at initialization time, then if that icon
+font hasn't loaded it may report the intrinsic width/height incorrectly. In order to prevent this,
+you can override the default behavior of `getClientRect()` to return the correct results. For
+example, if you know an icon font sizes its elements to `24px` width/height, you can do the
+following:
+
+```js
+this.ripple_ = new MDLRippleFoundation({
+  // ...
+  computeBoundingRect: () => Object.assign({}, element.getBoundingClientRect(), {
```

Comment:
My first thought was that this would yield incorrect `right` and `bottom` values in the resulting rect, but after testing it out, it looks like ClientRect's fields are not enumerable, so you end up with `{ width: 24, height: 24 }`.

http://codepen.io/shyndman/pen/bwpPEq?editors=0011